py3status/modules/yubikey.py

```diff
@@ -0,0 +1,183 @@
+# -*- coding: utf-8 -*-
+"""
+Show an indicator when YubiKey is waiting for a touch.
+
+Configuration parameters:
+    format: Display format for the module.
+        (default '\?if=is_waiting YubiKey')
+    gpg_check_timeout: Time to wait for gpg check response.
+        Use value as small as possible that doesn't introduce false positives.
+        (default 0.1)
+    gpg_check_watch_paths: A list of files to watch, start gpg check if any one was opened.
+        (default ['~/.gnupg/pubring.kbx', '~/.ssh/known_hosts'])
+    u2f_check_watch_paths: A list of files to watch, toggle u2f check if any one was opened.
+        (default ['~/.config/Yubico/u2f_keys'])
+
+Control placeholders:
+    {is_waiting} a boolean indicating whether YubiKey is waiting for a touch.
+
+SAMPLE OUTPUT
+{'color': '#00FF00', 'full_text': 'YubiKey'}
+
+Dependencies:
+    gpg: to check for pending gpg access request
+    inotify-simple: to check for pending gpg and u2f requests
+    github.com/maximbaz/pam-u2f: a fork that adds watch capability to pam-u2f module
+    subprocess32: if after all these years you are still using python2
+
+@author Maxim Baz (https://github.com/maximbaz)
+@license BSD
+"""
+
+import time
+import os
+import signal
+import sys
+import threading
+import inotify_simple
+
+if os.name == 'posix' and sys.version_info[0] < 3:
+    import subprocess32 as subprocess
+else:
+    import subprocess
```

Comment:
We import `sys` for this. Wanting to see if we can live w/o `sys`.
```
try:
    import subprocess  # py3
except:
    import subprocess32 as subprocess  # py2
```

Reply:
Not really, because `subprocess` exists in both py2 and py3, it's just that the module in py2 doesn't contain timeout functionality that was introduced in py3.

Reply:
Flip the `import subprocess` lines could be sufficent to omit `sys`... then I wonder if we should explicitly throw an exception on subprocess32 not imported and is python2... `yubikey: subprocess32 not installed`. py3status support python2 until or after https://pythonclock.org and I think `py3status` is a wordplay on `i3wm` which is also a wordplay on `wmii`. ;-)

Reply:
The exception will happen with current implementation, right? Python3 will import subprocess and nothing else, python2 will import subprocess32 and throw an exception if it cannot do so.

Reply:
>The exception will happen with current implementation, right?

Yup. Sharing the findings.

Packages: `inotify_simple`, `subprocess32`
```
2017-10-17 INFO loading module "yubikey" from .../yubikey.py
2017-10-17 INFO Module `yubikey` could not be loaded
```
No exceptions... The bar will report:
```
yubikey // yubikey: Import Error, No module named 'inotify_simple'
yubikey // yubikey: Import Error, No module named 'subprocess32'
```
It's just a matter of if we should need `sys`... and/or if we should make this nice like `module: not installed`... eg `yubikey: subprocess32 not imported` or `yubikey: inotify_simple not imported`. Anyhow, I leave this to others. Same with the formatting.